README.md

```diff
@@ -1,2 +1,478 @@
-# zartan
+# Zartan
 Web UI to create and manage http proxies from one or more sources.
+
+Web scrapers, crawlers or spiders are programs designed to automatically
+go to one or more websites and download content from them.  A well-behaved
+scraper will read that site's robots.txt file to determine which pages its
+allowed to read.  Not all scrapers are well-behaved.  Often a website
+has valuable information available to the public that the maintainers of
+that site only want humans to use.
+
+If you write a scraper targeting one such website then the second line of
+defense against your scraper is to blacklist your server's IP address.
+In order to continue scraping you need to change your apparent IP address
+frequently.  One such way to do this is to use HTTP proxies.
+
+Zartan will automatically create proxies from one or more sources and serve
+them to your web scraper through a RESTful API.  Zartan will maintain separate
+pools of proxies for each website you scrape, allowing you to use the same
+proxies to scrape multiple sites, even if specific proxies have been banned
+on specific sites.
+
+It is expected that you run Zartan on your own server.  This is not a web
+service.
+
+# API Usage
+All of the below GET and PUT requests require an api_key parameter.  The
+responses are all JSON.  If the api_key is missing or invalid, the response is:
+```
+{"result":"error","reason":"Unrecognized API Key"}
+```
+Go to the admin panel, generate
+an api_key and use that when making further requests.
+
+## Get a new proxy
+```
+GET http://HOST_NAME/v1/SITE_NAME
+```
+Parameters
+- older_than (optional)
+  - How many seconds a proxy must remain idle before being used again
+
+### On success
+```
+{
+  "result":"success",
+  "payload":{
+    "id":11,
+    "host":"IP_ADDRESS",
+    "port":PORT_NUMBER,
+    "source_id":2,
+    "deleted_at":null,
+    "created_at":"2015-03-10T15:57:17.014Z",
+    "updated_at":"2015-04-07T18:14:55.850Z"
+  }
+}
+```
+
+### Failure: No proxies available
+```
+{"result":"please_retry","interval":20}
+```
+This either happens when there are no proxies newer than the optional older_than
+parameter, or there are no proxies available at all.  In both cases, the
+client is asked to wait `interval` seconds before they make another request.
+
+## Report proxy success
+```
+PUT http://HOST_NAME/v1/SITE_NAME/11/succeeded
+```
+Informs zartan that the proxy with id #11 has successfully scraped a page.
```

Comment:
Since you have other variables in the URL, I'd recommend using PROXY_ID or similar. If you're concerned about making these look closer to the actual URLs, maybe provide an example like
```
PUT http://zartan.example.com/v1/example.org/11/succeeded
```